File: core/src/squares/components/TransformComponent.java
package squares.components;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.math.Vector2;

//Apache commons pool for pooling components.
public class TransformComponent implements Component {
    private Vector2 position;

    public TransformComponent() {
        this(Vector2.Zero);
    }

    public TransformComponent(Vector2 position) {
        this.position = position;
    }

    public Vector2 getPosition() {
        return position;
    }

    public void setPosition(Vector2 position) {
        this.position = position;
    }

    public float x() {
        return position.x;
    }

    public float y() {
        return position.y;
    }
}

File: core/src/com/mygdx/game/components/TransformComponent.kt
package com.mygdx.game.components

import com.badlogic.ashley.core.Component
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.math.Vector2
import com.badlogic.gdx.math.Vector3
import com.mygdx.game.algorithm.Point

/**
 * Class used to move character.
 */
class TransformComponent(x : Float, y : Float) : Component {
    val posit = Point(x.toInt(), y.toInt())
    val scale = Vector2(1.0f, 1.0f)
    var rotation = 0.0f
}
